drip_page_parser.js

```diff
@@ -0,0 +1,174 @@
+function parsePage() {
+  //run jQuery in compatibility mode as $jq
+  var $jq = window.jQuery;
+  $jq.noConflict();
+  $jq.fn.exists = function () {
+    if (this.length > 0) {
+      return this;
+    } else {
+      return false;
+    }
+  };
+
+  //define methods needed for parsing
+  var hexRegEx = new RegExp(/^#[0-9a-f]{3,6}$/i);
+  var rgb2hex = function (rgb) {
+    if (!rgb || rgb.indexOf("rgb") != 0) {
+      return rgb || "-";
+    }
+    var rgbArr = rgb.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
+
+    if (!rgbArr) {
+      rgbArr = rgb.match(/^rgba\((\d+),\s*(\d+),\s*(\d+),\s*(\d+)\)$/);
+    }
+    function hex(x) {
+      return ("0" + parseInt(x).toString(16)).slice(-2);
+    }
+    if (rgbArr && rgbArr[4] && rgbArr[4] == "0") {
+      return "transparent";
+    } else if (rgbArr) {
+      return "#" + hex(rgbArr[1]) + hex(rgbArr[2]) + hex(rgbArr[3]);
+    } else {
+      return "ERR";
+    }
+  };
+  var getTypeSet = function (el, name) {
+    return {
+      name: name,
+      "text-colour": rgb2hex(el.css("color") || naMsg),
+      font: el.css("font-family") || naMsg,
+      "font-size": el.css("font-size") || naMsg,
+      leading: el.css("line-height") || naMsg,
+      "font-style": el.css("font-style") || naMsg,
+    };
+  };
+  var orderColourResponse = function (colourArr, prependValueArr) {
+    colourArr = (colourArr || []).sort(function (a, b) {
+      return b[1] < a[1] ? -1 : b[1] > a[1] ? 1 : 0;
```

Comment:
```suggestion
      return b - a;
```

Maybe? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

Reply:
Stultify me stuff, not gonna touch it right now.